nodebuilder/p2p/pubsub.go

```diff
@@ -109,12 +110,23 @@ func topicScoreParams(network Network) map[string]*pubsub.TopicScoreParams {
 	return mp
 }
 
-func peerScoreParams(isBootstrapper bool, bootstrappers Bootstrappers) *pubsub.PeerScoreParams {
+func peerScoreParams(isBootstrapper bool, bootstrappers Bootstrappers, cfg Config) *pubsub.PeerScoreParams {
 	bootstrapperSet := map[peer.ID]struct{}{}
 	for _, b := range bootstrappers {
 		bootstrapperSet[b.ID] = struct{}{}
 	}
 
+	var ipColocFactWl []*net.IPNet = nil
+	if len(cfg.IPColocationFactorWhitelist) > 0 {
+		for _, strIp := range cfg.IPColocationFactorWhitelist {
+			_, ipNet, err := net.ParseCIDR(strIp)
+			if err != nil {
+				panic(err)
```

Comment:
We should not panic but return a nice error to the user, as it's clearly a misconfiguration by the user